### PR TITLE
Make UKRR table available

### LINKED ISSUE #2114: Make data in the `UKRR` table available in ehrQL
The "[Long-term complications after SARS-CoV-2 infection in relation to dialysis and kidney transplantation][1]" project has been granted access to data in the [`UKRR`][2] table.[^1] However, data in this table are not yet available in ehrQL. To allow the researcher (Viyaasan) to start working on their project, could data in this table be made available in ehrQL?

There is a separate issue for restricting the ehrQL table and for granting this project (or more correctly, a repository associated with this project) access to the ehrQL table (opensafely-core/opensafely-cli#270).

[1]: https://jobs.opensafely.org/long-term-complications-after-sars-cov-2-infection-in-relation-to-dialysis-and-kidney-transplantation/
[2]: https://reports.opensafely.org/reports/opensafely-tpp-database-schema/#UKRR
[^1]: https://bennettoxford.slack.com/archives/C31D62X5X/p1726755789454639

### PR DESCRIPTION
**Pull request by madwort**
* fixes #2114 
* use column naming & descriptions from cohortextractor

### CODE REVIEW

File: ehrql/backends/tpp.py [hunk 1]
```diff
@@ -1010,6 +1010,21 @@ def opa_proc(self):
...
+                renal_centre,
+                creat as latest_creatinine,
+                eGFR_ckdepi as latest_egfr,
+                rrt_start as rrt_start_date,
+                mod_start as treatment_modality_start,
+                mod_prev as treatment_modality_prevalence
+            FROM UKRR
+        """
+    )
+
```
Comment (evansd) [inline ehrql/backends/tpp.py:1027]: If you don't have to do any transformation of the columns then you can get a simpler definition, and cleaner resulting SQL, by using `MappedTable` like we do here:
https://github.com/opensafely-core/ehrql/blob/3757e09de8ba397e426f78c53fb6a0dbe0df7fd6/ehrql/backends/tpp.py#L252-L260

**Review by evansd — APPROVED** [review]
All looks great to me. Thanks!

As discussed on Slack though, we should double-check those categorical values still match the data before we deploy.

(And assuming you'll rebase away the typo fix before merging :smile:  )

File: ehrql/backends/tpp.py (hunk 1 above)
Comment (madwort) [reply ehrql/backends/tpp.py]: thanks, have done that

**Comment by madwort** [comment]
merged https://github.com/opensafely/tpp-database-categorical-columns/pull/4 but need to run & verify before merging this

**Comment by madwort** [comment]
ok, so, verified the results in TPP and I realise now that cohort-extractor was [mapping the values](https://github.com/opensafely-core/cohort-extractor/blob/eebf49133434c3b42a7dcca94dab8c423f6d54f7/cohortextractor/tpp_backend.py#L3474-L3480), e.g. `2019_prevalence` to `2019prev`

**Review by evansd — APPROVED** [review]
Looks great, thanks Tom.

**Pull request merged**